interface/javascript/general.js

```diff
@@ -1,207 +1,245 @@
-		domRefs.$stageContainer.style.top=window.stageSettings.posY+"px";
-		domRefs.$stageContainer.style.width=window.stageSettings.width+"px";
-		domRefs.$stageContainer.style.height=window.stageSettings.height+"px";
+		this.stageSettings.width=Math.round(this.stageSettings.width);
+		this.stageSettings.height=Math.round(this.stageSettings.height);
 		
-		domRefs.$pagination.style.right=(window.windowSize.width - window.stageSettings.posX - window.stageSettings.width)+"px";
+		this.domRefs.$stageContainer.style.left=this.stageSettings.posX+'px';
+		this.domRefs.$stageContainer.style.top=this.stageSettings.posY+'px';
+		this.domRefs.$stageContainer.style.width=this.stageSettings.width+'px';
+		this.domRefs.$stageContainer.style.height=this.stageSettings.height+'px';
 		
-		for(i=0; i<domRefs.$paginationItems.length; i++){
-			domRefs.$paginationItems[i].style.width=(window.windowSize.width - window.stageSettings.posX - window.stageSettings.width)+"px";
-			domRefs.$paginationItems[i].style.height=(window.windowSize.width - window.stageSettings.posX - window.stageSettings.width)+"px";
-		};
-	};
-	
-	const paginationClickListener = function(domRefs, loadAnimationDuration){
-		window.location.hash=this.getAttribute("href").split("#")[1];
-		removeClass(domRefs.$stage,"loaded");
-		setTimeout(function(){
-			location.reload();
-		}, loadAnimationDuration);
-		return false;		
-	};	
+		this.domRefs.$pagination.style.right=(this.windowSize.width - this.stageSettings.posX - this.stageSettings.width)+'px';
+		
+		this.domRefs.$paginationItems.forEach((el) => {
+			el.style.width=(this.windowSize.width - this.stageSettings.posX - this.stageSettings.width)+'px';
+			el.style.height=(this.windowSize.width - this.stageSettings.posX - this.stageSettings.width)+'px';
+		});
+
+		if(this.externalResizeListener != undefined){
+			this.externalResizeListener();
+		}
+	}
 	
-	const logoClickListener = function(domRefs, loadAnimationDuration){
-		window.location.hash=(window.currentStageID<animationsIDRep.length)?window.currentStageID+1:1;
-		removeClass(domRefs.$stage, "loaded");
-		setTimeout(function(){
-			location.reload();
-		}, loadAnimationDuration);
-		return false;
-	};
+	this.setStageAsLoaded = () => {
+		this.addClass(this.domRefs.$stage, 'loaded');
+	}
 	
 	
-	const stats=new Stats();
-	document.body.appendChild(stats.dom);
-	requestAnimationFrame(function loop(){
-		stats.update();
-		requestAnimationFrame(loop)
-	}
-	);
+	this.hasClass = (el, className) => {
+		return el.classList ? el.classList.contains(className) : new RegExp('\\b'+ className+'\\b').test(el.className);
+	};
+	this.addClass = (el, className) => {
+		if (el.classList) el.classList.add(className);
+		else if (!this.hasClass(el, className)) el.className += ' ' + className;
+	};
+	this.removeClass = (el, className) => {
+		if (el.classList) el.classList.remove(className);
+		else el.className = el.className.replace(new RegExp('\\b'+ className+'\\b', 'g'), '');
```

Comment:
why so complicated? `el.classList.add()` `el.classList.remove()` and `el.classList.toggle()`, no regex, no helper function needed:https://developer.mozilla.org/en-US/docs/Web/API/Element/classList